src/devTools.js

```diff
@@ -110,14 +119,16 @@ export function connect(options = {}) {
     unsubscribe: () => {
       delete listeners[id];
     },
-    send: (action, payload) => {
+    send: async (action, payload) => {
+      await obtainHostname(options, hostnamePromise);
       if (action) {
         send(action, payload, options, 'ACTION', id);
       } else {
         send(undefined, payload, options, 'STATE', id);
       }
     },
-    error: (payload) => {
+    error: async (payload) => {
+      await start(options, hostnamePromise);
```

Comment:
Wondering – why did we need to add this line? If we need to make sure we've resolved the `start` promise before calling `socket.emit`, we should make sure it's the same `start` promise as before, instead of having two of them.

We can assign my suggestion above to a variable:

```js
const startPromise = (async () => {
  await hostnamePromise;
  start(options);
})();
```

And then we can `await` that promise here:

```js
error: async (payload) => {
  await startPromise;
  socket.emit({ type: 'ERROR', payload, id: socket.id, instanceId: id });
}
```